File: objc/Builder/Builder/GeneralUsageComputerBuilder.m
#import "GeneralUsageComputerBuilder.h"

@implementation GeneralUsageComputerBuilder

-(void)buildCPU {
    [self.computer setCpu:@"Intel Core i5"];
}

-(void)buildMotherboard {
    [self.computer setMotherboardModel:@"8 GB"];
}

-(void)buildRAM {
    [self.computer setMotherboardModel:@"Gigabyte GA-H110M-S2"];
}

-(void)buildStorage {
    [self.computer setStorageValue:@"512 TB"];
}

-(void)buildGraphicsCard {
    [self.computer setGraphicsCardModel:@"Not installed"];
}


@end

File: swift/Builder/Builder/GeneralUsageComputerBuilder.swift
import Foundation

class GeneralUsageComputerBuilder: ComputerBuilder {
    override func buildCPU() {
        self.computer?.cpu = "Intel Core i5"
    }
    
    override func buildMotherboard() {
        self.computer?.motherboardModel = "Gigabyte GA-H110M-S2"
    }
    
    override func buildRAM() {
        self.computer?.ramValue = "8 GB"
    }
    
    override func buildStorage() {
        self.computer?.storageValue = "512 GB"
    }
    
    override func buildGraphicsCard() {
        self.computer?.graphicsCardModel = "Not installed"
    }
}
